src/detect-lua.c

```diff
@@ -474,9 +474,8 @@ static void *DetectLuaThreadInit(void *data)
 
     if (lua->allow_restricted_functions) {
         luaL_openlibs(t->luastate);
-    } else {
-        SCLuaSbLoadLibs(t->luastate);
     }
+    SCLuaSbLoadLibs(t->luastate);
```

Comment:
Keep if the if/else.  In the case of allowing restricted libs it should look like:
```
    luaL_openlibs(luastate);
    SCLuaRequirefBuiltIns(luastate);
```
As SCLuaSbLoadLibs is going to override the requires, plus re-load in already loaded libs, then wipe the ones that are not allowed. `SCLuaRequirefBuiltIns` just augments the standard libs with ours.